### PR TITLE
Bug 1178208 - Wrong time is displayed when the alarm rings after a ti…

### PR DESCRIPTION
**Pull request by mcav**
…me zone change.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=da5db713fe45936643bc7530705efcd3abe90ed4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d3bb1be638cf25ef0748f669b4eb1cf0de3a07c6

File: apps/clock/js/ring_view.js [hunk 1]
```diff
@@ -137,8 +137,25 @@ RingView.prototype = {
...
 
+    var displayedTime;
+    if (alert.type === 'alarm') {
+      // Display the time as specified with "hour" and "minute". The timestamp
+      // stored by mozAlarms does not move when the timezone changes, so we
+      // cannot just use the value stored in '.time'. See
+      // https://bugzilla.mozilla.org/show_bug.cgi?id=1178208#c4
+      displayedTime = new Date();
+      displayedTime.setHours(alert.hour);
+      displayedTime.setMinutes(alert.minute);
```
Comment (zbraniecki) [inline apps/clock/js/ring_view.js]: why not `displayedTime = Date.now()` ?

Comment (mcav) [inline apps/clock/js/ring_view.js]: The alarm UI might trigger long after the "scheduled alarm time" has passed, due to the phone being off or whatever. For UX reasons we display the time the alarm was supposed to go off, not the time we instantiated the dialog.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=43e215e9d34a8356ce39fe8043800310607f83ac

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=43e215e9d34a8356ce39fe8043800310607f83ac

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=068a69fddd1b7e260ea0da2ee0cb2993701c6ce3

**Pull request merged**